python/ngen_cal/src/ngen/cal/agent.py

```diff
@@ -57,9 +57,13 @@ def best_params(self) -> str:
 
 class Agent(BaseAgent):
 
-    def __init__(self, model_conf, workdir: 'Path', log: bool=False, restart: bool=False, parameters: 'Optional[Mapping[str, Any]]' = {}):
+    def __init__(self, model: Model, workdir: 'Path', log: bool=False, restart: bool=False, parameters: 'Optional[Mapping[str, Any]]' = {}):
         self._workdir = workdir
         self._job = None
+        assert not isinstance(model.model, NoModel), "invariant"
+        # NOTE: if support for new models is added, this will need to be modified
+        model_inner = model.model.unwrap()
```

Comment:
It may be worth a quick description of what the difference between `model_inner` and `model` is here

Reply:
Yeah that was a pretty awful name haha. Ill change that